commit c85ddca16c757409d3e589b5808679fd81af6b58
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2018-08-06T22:45:42+00:00 (1 year, 2 months ago)
    
    TRITON-651 "sysinfo-refresh" job needs to be more careful about what NIC fields it updates

diff --git a/lib/workflows/server-sysinfo.js b/lib/workflows/server-sysinfo.js
index 40e3e11..293cf35 100644
--- a/lib/workflows/server-sysinfo.js
+++ b/lib/workflows/server-sysinfo.js
@@ -242,8 +242,6 @@ function getServerNics(job, cb) {
                 = job.params.existingNics[n];
         }
 
-        var listEqual;
-
         for (n in sysinfoNics) {
             var sysinfoNic = sysinfoNics[n];
             var napiNic = napiNics[sysinfoNic['MAC Address']];
@@ -298,36 +296,6 @@ function getServerNics(job, cb) {
                 newNic.belongs_to_type = 'server';
             }
 
-            // This can't be declared outside of this function, otherwise it
-            // can't be seen when running in a workflow
-            listEqual = function (a, b) {
-                if (!a && !b) {
-                    return true;
-                }
-
-                if (!a || !b || (a.length !== b.length)) {
-                    return false;
-                }
-
-                a.sort();
-                b.sort();
-
-                for (var i = 0; i < a.length; i++) {
-                    if (a[i] !== b[i]) {
-                        return false;
-                    }
-                }
-
-                return true;
-            };
-
-            var equal =
-                listEqual(sysinfoNic['NIC Names'], napiNic.nic_tags_provided);
-
-            if (sysinfoNic.hasOwnProperty('NIC Names') && !equal) {
-                newNic.nic_tags_provided = sysinfoNic['NIC Names'];
-            }
-
             if (Object.keys(newNic).length !== 0) {
                 newNic.mac = sysinfoNic['MAC Address'];
                 toUpdateNics.push(newNic);
